app/assets/v2/js/cart-ethereum-zksync.js

```diff
@@ -111,6 +119,13 @@ Vue.component('grantsCartEthereumZksync', {
       appCart.$refs.cart.handleError(e);
     },
 
+    // Wait for correct chainId
+    waitForCorrectChain() {
+      return new Promise((resolve, _reject) => {
+        this.stopWaitingForNetwork = resolve;
```

Comment:
So we create a new `promise` every time `onChangeHandler` is called and then we resolve it when `networkId` is called - but to what end? I'm struggling to understand what this is trying to do? Couldn't we just rely on `networkId` as a global?

Reply:
@gdixon  We ar enot waiting for the `networkId`, we are actually waiting for the change of networkId to be propagated to the `cart.js` which in turn will update the address in the Gitcoin grant.
The way it is build is misleading I have to admit ... 
Maybe the cleanest way to handle is would be to update the address of the Gitcoin grant in `cart-ethereum-zksync.js`, in the checkout function. Then this logic would not span across multiple files at least?